Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -8477,6 +8481,12 @@ MediaProducerMediaStateFlags HTMLMediaElement::mediaState() const
     UNUSED_VARIABLE(hasAudio);
 #endif
 
+#if ENABLE(MEDIA_SOURCE)
+    // We can assume that if we have active source buffers, later networking activity (such as stream or XHR requests) will be media related.
+    if (m_mediaSource && m_mediaSource->activeSourceBuffers() && m_mediaSource->activeSourceBuffers()->length())
```

Comment:
a source buffer becomes active when an init segment has been added, this signals that more data is to come.

it would lead to faster start time with players such as YouTube than waiting to actual get data and sufficient buffer for playback to start.

Reply:
An alternative would be to trigger the signal when a media element is tied to a media source even though not yet playing. This would be even sooner, but maybe too aggressive with existing websites?